lib/wallet/http.js

```diff
@@ -833,6 +784,473 @@ class HTTP extends Server {
       await req.wallet.resend();
       res.json(200, { success: true });
     });
+
+    // Wallet Name States
+    this.get('/wallet/:id/name', async (req, res) => {
+      const height = this.wdb.height;
+      const network = this.network;
+
+      const names = await req.wallet.getNames();
+      const items = [];
+
+      for (const ns of names)
+        items.push(ns.getJSON(height, network));
+
+      res.json(200, items);
+    });
+
+    // Wallet Name State
+    this.get('/wallet/:id/name/:name', async (req, res) => {
+      const valid = Validator.fromRequest(req);
+      const name = valid.str('name');
+
+      assert(name, 'Must pass name.');
+      assert(rules.verifyName(name), 'Must pass valid name.');
+
+      const height = this.wdb.height;
+      const network = this.network;
+      const ns = await req.wallet.getNameStateByName(name);
+
+      res.json(200, ns.getJSON(height, network));
+    });
+
+    // Wallet Auction
+    this.get('/wallet/:id/auction/:name', async (req, res) => {
+      const valid = Validator.fromRequest(req);
+      const name = valid.str('name');
+
+      const height = this.wdb.height;
+      const network = this.network;
+
+      assert(name, 'Must pass name.');
+      assert(rules.verifyName(name), 'Must pass valid name.');
+
+      const ns = await req.wallet.getNameStateByName(name);
+
+      if (!ns)
+        return res.json(404);
+
+      const bids = await req.wallet.getBidsByName(name);
+      const reveals = await req.wallet.getRevealsByName(name);
+
+      const info = ns.getJSON(height, network);
+      info.bids = [];
+      info.reveals = [];
+
+      for (const bid of bids)
+        info.bids.push(bid.toJSON());
+
+      for (const reveal of reveals)
+        info.reveals.push(reveal.toJSON());
+
+      return res.json(200, info);
+    });
+
+    // All Wallet Bids
+    this.get('/wallet/:id/bid', async (req, res) => {
```

Comment:
`GET /wallet/:id/bid` will return all of the bids for the wallet between all of the names. `GET /wallet/:id/bid/:name` will return all of the bids for a given name. Right now, the client has a method for each endpoint

- `getBid` for `/wallet/:id/bid/`
- `getBids` for `/wallet/:id/bid/:name`

Both endpoints can return multiple bids, so the naming is not ideal. I think it could be `getBids` and `getBidsByName` but then its getting really verbose. Thoughts?

The same thing applies to the reveal endpoints.

Reply:
Decided to migrate to `getBidsByName` and `getRevealsByName` because those exist as methods on the `wallet` itself.